src/js/main.js

```diff
@@ -0,0 +1,340 @@
+            document.querySelector('.textareaEl').selectionEnd = result;
+          }
+        })();
+        break;
+      default:
+        break;
+    }
+  }
+
+  keysUpHandling(value) {
+    if (this.buffer.indexOf('Alt') !== -1 && this.buffer.indexOf('Shift') !== -1) {
+      this.updatelayout();
+    }
+    switch (value.querySelector('.main').innerHTML) {
+      case 'Shift':
+        this.updateCaps();
+        break;
+      default:
+        break;
+    }
+    if (this.buffer.indexOf(value.querySelector('.main').innerHTML) !== -1) {
+      this.buffer.splice(this.buffer.indexOf(value.querySelector('.main').innerHTML), 1);
+    }
+  }
+
+  setLayoutStorage(value) {
+    if (value) {
+      localStorage.removeItem('lang');
+      localStorage.setItem('lang', `${value}`);
+    } else if (!localStorage.getItem('lang')) {
+      localStorage.removeItem('lang');
+      localStorage.setItem('lang', `${this.lang}`);
+    } else {
+      this.lang = localStorage.getItem('lang');
+    }
+  }
+
+  printValue(value) {
+    this.indexCursor = document.querySelector('.textareaEl').selectionStart;
+    const tempStr = document.querySelector('.textareaEl').innerHTML;
+    this.partString = tempStr.slice(0, this.indexCursor) + value + tempStr.slice(this.indexCursor);
+    document.querySelector('.textareaEl').innerHTML = this.partString;
+    document.querySelector('.textareaEl').selectionStart = this.indexCursor + 1;
+    document.querySelector('.textareaEl').selectionEnd = this.indexCursor + 1;
+  }
+
+  static pushDownAnim(param) {
+    const value = param;
+    Keyboard.animateButton({
+      duration: 100,
+      timing: (timeFraction) => timeFraction ** 2,
+      draw: (progress) => {
+        value.style.top = `-${3 * progress}px`;
+        value.style.left = `-${3 * progress}px`;
+        value.style.boxShadow = `${4 * progress - 4}px ${4 * progress - 4}px 4px 2px #000000`;
+      },
+    });
+  }
+
+  static pushUpAnim(param) {
+    const value = param;
+    Keyboard.animateButton({
+      duration: 100,
+      timing: (timeFraction) => timeFraction ** 2,
+      draw: (progress) => {
+        value.style.top = `${0 * progress}px`;
+        value.style.left = `${0 * progress}px`;
+        value.style.boxShadow = `${0 * progress - 4}px ${0 * progress - 4}px 4px 2px #000000`;
+      },
+    });
+  }
+
+  static animateButton(options) {
+    const start = performance.now();
+    requestAnimationFrame(function animate(time) {
+      // timeFraction от 0 до 1
+      let timeFraction = (time - start) / options.duration;
+      if (timeFraction > 1) timeFraction = 1;
+      // текущее состояние анимации
+      const progress = options.timing(timeFraction);
+      options.draw(progress);
+      if (timeFraction < 1) {
+        requestAnimationFrame(animate);
+      }
+    });
+  }
+}
+
+const keyboard = new Keyboard();
+
+const LoadMethods = new Promise((resolve, reject) => {
+  if (Keyboard.addHtml()) { resolve(); } else { reject(); }
+});
```

Comment:
To my mind, it's too complex logic of handling errors as the code runs synchronously. How about try/catch?